R/matrixfit.R

```diff
@@ -561,8 +561,8 @@ plot.matrixfit <- function(mfit, plot.errorband=FALSE, ylim, xlab="t/a", ylab="y
     if(mfit$model == "shifted") y <- pars[2]*pars[3]*( exp(-pars[1]*(tx-deltat/2)) - sgn*exp(-pars[1]*(T-(tx-deltat/2))))
     else if(mfit$model == "pc") y <- pcModel(par=par[1:3], t=tx, T=T, reference_time=mfit$reference_time)
     else y <- 0.5*pars[2]*pars[3]*( exp(-pars[1]*tx) + sgn*exp(-pars[1]*(T-tx)))
-    yp <- rep(1, times=length(tt))
     
+    # yp is the physical exponential in case we want to look at the ratio plot
```

Comment:
does this work? shouldn't yp exist in the scope outside of the if statement or does it already and I just don't see it here?

Reply:
I am not sure what exactly you mean. R has function scope, not lexical scope. So we can define variables within an `if`-statement, even though that would not work in C++.

Also this block is the first and only definition of `yp` in that function (in vim: call `:set hlsearch`, put the cursor onto it and press `*` or just search `/\v<yp>`). So I think that the `else` was indeed needed here.

Reply:
You're right of course, but why have I had this fail before?

Reply:
yp was defined further above and then overwritten in the original implementation, as is often done in hadron

Reply:
As long as it works I'm happy.

Reply:
> `yp` was defined further above and then overwritten in the original implementation, as is often done in hadron.

I see the point of this style of programming in C/C++, but when I have function scope I prefer to do it this way. As it is here I read it as “`yp` is this or that.”, the other style reads as “`yp` is that. Except in this case.”. In C++ I really like `auto const yp = … ? … : …;` just because I can put a `const` into it. I guess my Haskell excursion made me think of variables as constants and now I want to avoid changing values.